### ISSUE TITLE
Do you train LSTMN model using Theano scan op or unroll scan?

### ISSUE THREAD

**Issue by junfenglx** [SourKream/NLP-Project#1]
Hello SourKream:

I read your LSTMN implementation, it's a good work.
I want to know that do you train LSTMN model using tensorflow or theano?
If you use theano, do you use unroll scan with padding sentences or  theano scan op?

Please let me know.
Thanks.

**Comment by SourKream** [comment]
Hi
- I was using Theano backend with Keras.
- I'm not very familiar with theano (hence the need for using Keras) but I think how keras does it is by using the theano scan function. I modified the step function in the recurrent class to adapt the layer according to the paper. 

P.S. Sorry for the directory being a huge mess right now. I would be cleaning it up soon.

**Comment by codedecde** [comment]
Keras uses the theano scan function by default (which is also the one that
we used).
The unroll scan is also supported in keras. However when we were working on
the project, it was buggy.

**Comment by junfenglx** [comment]
@SourKream @codedecde 
Thank you for replay.

Because scan can't get all previous state at current step t, so I post a issue
https://github.com/Theano/Theano/issues/4752 on Theano, **nouiz** tell me I
can't use global variable to save output state, must use output variable to save states,
I also tried the concatenation way, but didn't not success.
So I post the issue.

Anyway, Thank you for telling me the answer.
Maybe it's time for me to learn Keras how to
define custom layers, previous time I use Lasagne.
